pkg/webhook/webhook.go

```diff
@@ -257,7 +255,7 @@ func (w *Webhook) Handle(ctx context.Context, req admission.Request) admission.R
 		kupidRequestsTotal.With(prometheus.Labels{labelType: typeError}).Inc()
 		return admission.Allowed(err.Error())
 	} else if !mutated {
-		l.V(1).Info("Nothing mutated", "obj", obj)
+		l.Info("Nothing mutated for request")
```

Comment:
I deployed kupid newer version locally and I see now the logs are much improved.
However I felt this can create a lot of noise as when I deleted the affinity from etcd-main sts.. the logs showed following though I see 2 log entries ideally one should be enough.
```
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680062807.091456,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"statefulsets"},"namespace":"shoot--dev--ashlcl-nha","name":"etcd-main"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680062807.094295,"logger":"webhook","msg":"Mutated response for request","resource":{"group":"apps","version":"v1","resource":"statefulsets"},"namespace":"shoot--dev--ashlcl-nha","name":"etcd-main"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680062807.1265812,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"statefulsets"},"namespace":"shoot--dev--ashlcl-nha","name":"etcd-main"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680062807.1318276,"logger":"webhook","msg":"Mutated response for request","resource":{"group":"apps","version":"v1","resource":"statefulsets"},"namespace":"shoot--dev--ashlcl-nha","name":"etcd-main"}
```

After waiting for sometime the logs just added a few more 
```
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680062841.6576464,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"network-problem-detector-controller"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680062841.6591308,"logger":"webhook","msg":"Nothing mutated for request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"network-problem-detector-controller"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680062841.706419,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"network-problem-detector-controller"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680062841.7082741,"logger":"webhook","msg":"Nothing mutated for request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"network-problem-detector-controller"}
gardener-extension-kupid-98978b5b8-vrfn9 {"level":"info","ts":1680062961.5855594,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"garden","name":"nginx-ingress-controller"}
gardener-extension-kupid-98978b5b8-vrfn9 {"level":"info","ts":1680062961.5870636,"logger":"webhook","msg":"Nothing mutated for request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"garden","name":"nginx-ingress-controller"}
gardener-extension-kupid-98978b5b8-vrfn9 {"level":"info","ts":1680062961.613356,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"garden","name":"nginx-ingress-controller"}
gardener-extension-kupid-98978b5b8-vrfn9 {"level":"info","ts":1680062961.6145413,"logger":"webhook","msg":"Nothing mutated for request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"garden","name":"nginx-ingress-controller"}
gardener-extension-kupid-98978b5b8-vrfn9 {"level":"info","ts":1680062961.6278121,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"garden","name":"nginx-ingress-k8s-backend"}
gardener-extension-kupid-98978b5b8-vrfn9 {"level":"info","ts":1680062961.6287944,"logger":"webhook","msg":"Nothing mutated for request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"garden","name":"nginx-ingress-k8s-backend"}
```

If just run reconcile of a shoot then this becomes even more noisy and even captures all resources across all namespaces. This looked counter productive to me to search for 3 or 4 resources which are part of the scheduling policies amongst 398 pods just for a seed and one shoot in it.
<details><summary> Expand to see the noise you will see extensions like network-filter also being part of it </summary>

```yaml
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063355.9291072,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"statefulsets"},"namespace":"shoot--dev--ashlcl-nha","name":"etcd-events"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063355.9297528,"logger":"webhook","msg":"Nothing mutated for request","resource":{"group":"apps","version":"v1","resource":"statefulsets"},"namespace":"shoot--dev--ashlcl-nha","name":"etcd-events"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063355.9330943,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"statefulsets"},"namespace":"shoot--dev--ashlcl-nha","name":"etcd-main"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063355.934937,"logger":"webhook","msg":"Mutated response for request","resource":{"group":"apps","version":"v1","resource":"statefulsets"},"namespace":"shoot--dev--ashlcl-nha","name":"etcd-main"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063355.998925,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"statefulsets"},"namespace":"shoot--dev--ashlcl-nha","name":"etcd-main"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063356.0007033,"logger":"webhook","msg":"Mutated response for request","resource":{"group":"apps","version":"v1","resource":"statefulsets"},"namespace":"shoot--dev--ashlcl-nha","name":"etcd-main"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063356.104225,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"statefulsets"},"namespace":"shoot--dev--ashlcl-nha","name":"etcd-events"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063356.104841,"logger":"webhook","msg":"Nothing mutated for request","resource":{"group":"apps","version":"v1","resource":"statefulsets"},"namespace":"shoot--dev--ashlcl-nha","name":"etcd-events"}
gardener-extension-kupid-98978b5b8-hlhds {"level":"info","ts":1680063355.9836373,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"statefulsets"},"namespace":"shoot--dev--ashlcl-nha","name":"etcd-events"}
gardener-extension-kupid-98978b5b8-hlhds {"level":"info","ts":1680063355.98466,"logger":"webhook","msg":"Nothing mutated for request","resource":{"group":"apps","version":"v1","resource":"statefulsets"},"namespace":"shoot--dev--ashlcl-nha","name":"etcd-events"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063356.117172,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"statefulsets"},"namespace":"shoot--dev--ashlcl-nha","name":"etcd-main"}
gardener-extension-kupid-98978b5b8-hlhds {"level":"info","ts":1680063356.061197,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"statefulsets"},"namespace":"shoot--dev--ashlcl-nha","name":"etcd-main"}
gardener-extension-kupid-98978b5b8-hlhds {"level":"info","ts":1680063356.0645607,"logger":"webhook","msg":"Mutated response for request","resource":{"group":"apps","version":"v1","resource":"statefulsets"},"namespace":"shoot--dev--ashlcl-nha","name":"etcd-main"}
gardener-extension-kupid-98978b5b8-hlhds {"level":"info","ts":1680063356.097266,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"statefulsets"},"namespace":"shoot--dev--ashlcl-nha","name":"etcd-events"}
gardener-extension-kupid-98978b5b8-hlhds {"level":"info","ts":1680063356.0977602,"logger":"webhook","msg":"Nothing mutated for request","resource":{"group":"apps","version":"v1","resource":"statefulsets"},"namespace":"shoot--dev--ashlcl-nha","name":"etcd-events"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063356.118913,"logger":"webhook","msg":"Mutated response for request","resource":{"group":"apps","version":"v1","resource":"statefulsets"},"namespace":"shoot--dev--ashlcl-nha","name":"etcd-main"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063356.2154138,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"statefulsets"},"namespace":"shoot--dev--ashlcl-nha","name":"etcd-main"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063356.2174034,"logger":"webhook","msg":"Mutated response for request","resource":{"group":"apps","version":"v1","resource":"statefulsets"},"namespace":"shoot--dev--ashlcl-nha","name":"etcd-main"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063356.2393627,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"statefulsets"},"namespace":"shoot--dev--ashlcl-nha","name":"etcd-main"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063356.2412705,"logger":"webhook","msg":"Mutated response for request","resource":{"group":"apps","version":"v1","resource":"statefulsets"},"namespace":"shoot--dev--ashlcl-nha","name":"etcd-main"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063361.1567512,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"aws-custom-route-controller"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063361.1574147,"logger":"webhook","msg":"Nothing mutated for request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"aws-custom-route-controller"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063361.362221,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"aws-custom-route-controller"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063361.3628705,"logger":"webhook","msg":"Nothing mutated for request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"aws-custom-route-controller"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063361.4108713,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"cloud-controller-manager"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063361.411871,"logger":"webhook","msg":"Nothing mutated for request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"cloud-controller-manager"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063361.4964132,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"cloud-controller-manager"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063361.4971266,"logger":"webhook","msg":"Nothing mutated for request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"cloud-controller-manager"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063361.5514162,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"csi-driver-controller"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063361.552579,"logger":"webhook","msg":"Nothing mutated for request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"csi-driver-controller"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063361.64691,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"csi-driver-controller"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063361.6476803,"logger":"webhook","msg":"Nothing mutated for request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"csi-driver-controller"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063361.6745846,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"csi-snapshot-controller"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063361.6750033,"logger":"webhook","msg":"Nothing mutated for request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"csi-snapshot-controller"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063361.727344,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"csi-snapshot-controller"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063361.7278361,"logger":"webhook","msg":"Nothing mutated for request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"csi-snapshot-controller"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063361.7614584,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"csi-snapshot-validation"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063361.7619355,"logger":"webhook","msg":"Nothing mutated for request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"csi-snapshot-validation"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063361.8437595,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"csi-snapshot-validation"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063361.8443604,"logger":"webhook","msg":"Nothing mutated for request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"csi-snapshot-validation"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063370.2477283,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"kube-apiserver"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063370.2491822,"logger":"webhook","msg":"Nothing mutated for request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"kube-apiserver"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063370.3337433,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"kube-apiserver"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063370.335047,"logger":"webhook","msg":"Nothing mutated for request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"kube-apiserver"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063371.7534633,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"kube-controller-manager"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063371.754096,"logger":"webhook","msg":"Nothing mutated for request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"kube-controller-manager"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063371.848741,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"kube-controller-manager"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063371.849674,"logger":"webhook","msg":"Nothing mutated for request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"kube-controller-manager"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063371.9524348,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"vpn-seed-server"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063371.953478,"logger":"webhook","msg":"Nothing mutated for request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"vpn-seed-server"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063372.1231833,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"vpn-seed-server"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063372.1238773,"logger":"webhook","msg":"Nothing mutated for request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"vpn-seed-server"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063373.2374268,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"gardener-resource-manager"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063373.2380927,"logger":"webhook","msg":"Nothing mutated for request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"gardener-resource-manager"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063373.3208163,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"gardener-resource-manager"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063373.321768,"logger":"webhook","msg":"Nothing mutated for request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"gardener-resource-manager"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063375.0500014,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"kube-scheduler"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063375.0509965,"logger":"webhook","msg":"Nothing mutated for request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"kube-scheduler"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063375.107942,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"kube-scheduler"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063375.109373,"logger":"webhook","msg":"Nothing mutated for request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"kube-scheduler"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063375.7913227,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"vpa-admission-controller"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063375.7918305,"logger":"webhook","msg":"Nothing mutated for request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"vpa-admission-controller"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063375.8669322,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"vpa-admission-controller"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063375.867494,"logger":"webhook","msg":"Nothing mutated for request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"vpa-admission-controller"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063376.8286233,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"vpa-recommender"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063376.829217,"logger":"webhook","msg":"Nothing mutated for request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"vpa-recommender"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063376.9348938,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"vpa-recommender"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063376.9354694,"logger":"webhook","msg":"Nothing mutated for request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"vpa-recommender"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063377.4436352,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"vpa-updater"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063377.4441948,"logger":"webhook","msg":"Nothing mutated for request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"vpa-updater"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063377.5548303,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"vpa-updater"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063377.5555565,"logger":"webhook","msg":"Nothing mutated for request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"vpa-updater"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063381.338128,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"machine-controller-manager"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063381.33919,"logger":"webhook","msg":"Nothing mutated for request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"machine-controller-manager"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063381.4406428,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"machine-controller-manager"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063381.4416957,"logger":"webhook","msg":"Nothing mutated for request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"machine-controller-manager"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063381.574256,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"cluster-autoscaler"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063381.5750556,"logger":"webhook","msg":"Nothing mutated for request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"cluster-autoscaler"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063381.6757789,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"istio-system","name":"istiod"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063381.6764781,"logger":"webhook","msg":"Nothing mutated for request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"istio-system","name":"istiod"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063381.6917899,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"istio-system","name":"istiod"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063381.6925452,"logger":"webhook","msg":"Nothing mutated for request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"istio-system","name":"istiod"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063387.3195136,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"cluster-autoscaler"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063387.3200517,"logger":"webhook","msg":"Nothing mutated for request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"cluster-autoscaler"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063387.4668832,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"cluster-autoscaler"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063387.4679813,"logger":"webhook","msg":"Nothing mutated for request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"shoot--dev--ashlcl-nha","name":"cluster-autoscaler"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063399.2740295,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"extension-extension-shoot-networking-filter-j5w6g","name":"gardener-extension-shoot-networking-filter"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063399.2753606,"logger":"webhook","msg":"Nothing mutated for request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"extension-extension-shoot-networking-filter-j5w6g","name":"gardener-extension-shoot-networking-filter"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063399.3118057,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"extension-extension-shoot-networking-filter-j5w6g","name":"gardener-extension-shoot-networking-filter"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063399.3128207,"logger":"webhook","msg":"Nothing mutated for request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"extension-extension-shoot-networking-filter-j5w6g","name":"gardener-extension-shoot-networking-filter"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063399.3502443,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"extension-reserve-excess-capacity-4rhcn","name":"reserve-excess-capacity-etcd"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063399.351476,"logger":"webhook","msg":"Nothing mutated for request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"extension-reserve-excess-capacity-4rhcn","name":"reserve-excess-capacity-etcd"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063399.360577,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"extension-reserve-excess-capacity-4rhcn","name":"reserve-excess-capacity-etcd"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063399.361669,"logger":"webhook","msg":"Nothing mutated for request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"extension-reserve-excess-capacity-4rhcn","name":"reserve-excess-capacity-etcd"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063419.1449754,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"daemonsets"},"namespace":"extension-provider-aws-lt6rw","name":"mtu-customizer"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063419.1461754,"logger":"webhook","msg":"Nothing mutated for request","resource":{"group":"apps","version":"v1","resource":"daemonsets"},"namespace":"extension-provider-aws-lt6rw","name":"mtu-customizer"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063419.1576722,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"extension-provider-aws-lt6rw","name":"gardener-extension-provider-aws"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063419.1591263,"logger":"webhook","msg":"Nothing mutated for request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"extension-provider-aws-lt6rw","name":"gardener-extension-provider-aws"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063419.1728058,"logger":"webhook","msg":"Handling request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"extension-provider-aws-lt6rw","name":"gardener-extension-provider-aws"}
gardener-extension-kupid-98978b5b8-pdlc8 {"level":"info","ts":1680063419.1740403,"logger":"webhook","msg":"Nothing mutated for request","resource":{"group":"apps","version":"v1","resource":"deployments"},"namespace":"extension-provider-aws-lt6rw","name":"gardener-extension-provider-aws"}
```
</details>
So I think we should relook at the logLevel for this one and only use it as a debug mode to see if it was a false positive. wdyt?

Reply:
If we change the log level to debug level, it's not of much use when we run kupid productively because the debug level logs are never emitted and thus never captured by Loki.

I can definitely remove the noisy `Mutated response for request` and `Nothing mutated for request` logs, but I would vote for retaining the `Handling request` logs as these are essential in figuring out which requests were handled by Kupid.

Reply:
Ideally, Kupid shouldn't be handling every request in the cluster but only the ones relevant to the objects it wishes to mutate, based on the PSPs and CPSPs. We should ideally be using something like https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#matching-requests-objectselector to filter out the requests handled by the webhook in the first place. But this needs some thought and effort, and our first prio is to improve logging for better debugging.

I'll capture this info in an issue.

Reply:
https://github.com/gardener/kupid/issues/55

Reply:
The same link also states the following:
```
Use the object selector only if the webhook is opt-in, because end users may skip the admission webhook by setting the labels.
```
So one must be careful if that needs to be prevented.